test/integration/server/helpers/helpers.go

```diff
@@ -47,22 +47,32 @@ func CreateRepo(ctx context.Context, gp gitprovider.Client, url string) (gitprov
 		return nil, ref, fmt.Errorf("error parsing url: %w", err)
 	}
 
+	defaultBranch := "main"
 	repo, _, err := gp.OrgRepositories().Reconcile(ctx, *ref, gitprovider.RepositoryInfo{
 		Description:   gitprovider.StringVar("Integration test repo"),
 		Visibility:    gitprovider.RepositoryVisibilityVar(gitprovider.RepositoryVisibilityPrivate),
-		DefaultBranch: gitprovider.StringVar("main"),
+		DefaultBranch: gitprovider.StringVar(defaultBranch),
 	}, &gitprovider.RepositoryCreateOptions{AutoInit: gitprovider.BoolVar(true)})
 
 	if err != nil {
 		return nil, nil, fmt.Errorf("could not reconcile org repo: %w", err)
 	}
 
-	err = utils.WaitUntil(bytes.NewBuffer([]byte{}), 3*time.Second, 5*time.Second, func() error {
-		_, err := gp.OrgRepositories().Get(ctx, *ref)
+	err = utils.WaitUntil(os.Stdout, 3*time.Second, 9*time.Second, func() error {
```

Comment:
:question: Why do we need to log this to `stdout`?

Reply:
This is going to help us identify the exact errors this function is facing before every retry and make sure it is retrying enough times. For instance, I didn't realize that we were retrying only once as a result of poll-time being 3sec and timeout being 5sec. I just noticed it until I made the change to `stdout`. As a result, I increased the timeout to 9 seconds so at least it retries 3 times. Another aspect I've seen on previous occasions is that it might hit different errors in each of the retries. Each of those errors could give us more information when debugging.

Reply:
Cool makes sense to me